### PR TITLE
Code style

### PR DESCRIPTION
**Pull request by adamdebono**
Created style guide, and updated source to conform

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Contributing
+Below are guidelines for contributing to the SnapIn API. These are guidelines, not hard rules, so use your best judgement and feel free to propose changes. Unless there is a good reason to, however, these will be enforced across pull requests and issues.
+
+## Code Style
+All code must conform to the guidelines in [STYLE.md](STYLE,md) (but feel free to propose reasonable changes to that).
```
Comment (Reidsy) [inline CONTRIBUTING.md]: [STYLE.md](STYLE.md)
you have a comma in where a period should be

File: STYLE.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
+# Style Guide
+
+
+## Whitespace
+Use hard tabs, not spaces. This allows everyone to set their own preference for tab width in their editor. Make sure to set this option in your editor's preferences to enforce this.
```
Comment (Reidsy) [inline STYLE.md]: Please set hard tab width to 4 characters (or whatever you prefer). Allowing users to set their own tab width can lead to cases where some code will not align. Enforcing 4 character tabs makes it easy to enforce alignment rules.

File: STYLE.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+Avoid splitting statements over multiple lines where possible.
+
+If you use [Atom](http://atom.io), disable the 'Remove Trailing Whitespace' option from the 'Whitespace' plugin.
+
```
Comment (Reidsy) [inline STYLE.md]: These plugins should be enabled.

- Trailing whitespace is bad and can sometimes cause merge conflicts
- Files should end with a trailing newline
- Empty lines should not have any indentation

File: STYLE.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+## Commenting
+Comments should be useful and accurate. Inline commenting should be limited to strange, complicated or hard to follow code. Documenting classes and methods are required. Always update comments to reflect code changes. All non-Javadoc code should use a double slash, even multi-line comments.
+
+Do not comment out 'dead' code and leave it there. Source control exists to keep previous versions of code, in case it is useful at a later date. Commented code is allowable when under active development.
+
+Inline comments should have a space after the double-slash. All comments should
+be wrapped at 80 characters for readability.
+
+### TODOs
+You may use ```TODO```, ```CHANGEME``` and ```FIXME``` in code to note things that need to be completed. These should never be found in live code. If you wish to mention someone, use an '@' followed by their github username.
```
Comment (Reidsy) [inline STYLE.md]: may be worth noting that code review will fail if any of these comments are caught. If there is something "TODO" please include it in the pull request, or create an issue for it.

I'm also okay with prefixing these TODOS with '@' as some editors detect this and display them in a fancy way

File: STYLE.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+### Javadoc
+Use Javadoc-style commenting to document all classes, methods and properties. Documentation is a requirement for each of these.
+
+To specify a Javadoc block, use an opening tag with two asterisks (```/**```), and place an asterisk at the start of each line in the block.
+
+#### Required Tags
+- ```@param``` The parameters for a method. Only required for method documentation.
+- ```@return``` The return value of a method. Only required for method documentation.
+- ```@author``` The person who created the object. This should be their full name, followed by they github username surrounded by '<>'.
```
Comment (Reidsy) [inline STYLE.md]: do we care if someone doesn't have/use github?

Comment (adamdebono) [inline STYLE.md]: If they don't use github then they can't contribute

File: STYLE.md (hunk 4 above)
Comment (adamdebono) [inline STYLE.md]: I have mentioned in CONTRIBUTING.md that pulls will fail for anything in the style guide.

I'd prefer not to use @ with TODOs, because @ is generally used for other things

File: STYLE.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
+# Style Guide
+
+
+## Whitespace
+Use hard tabs, not spaces. Please set your tab width to 4 characters (using something else can cause alignment issues). Make sure to set this option in your editor's preferences to enforce this.
+
+Trailing whitespace can (sometimes) be bad an cause merge conflicts, so remove all trailing whitespace (your editor should be able to do this for you automatically). This includes indentation on empty lines. All files should end with a newline.
```
Comment (Reidsy) [inline STYLE.md]: (sometimes) be bad **and** cause

typo. Then this is good to go
